### DISCUSSION TITLE
API

### DISCUSSION THREAD

**Discussion by MridulNegi2005** [Shabinder/SpotiFlyer#131] [General]
Please make a API in which we can send request and it returns data(like download link) in json format. I want to to make a bot for discord using discord.py. 
If API requests already exist,then please tell me.

**Comment by Shabinder** [comment]
Here you can find the api's I use : https://github.com/Shabinder/SpotiFlyer/tree/main/common/dependency-injection/src/commonMain/kotlin/com/shabinder/common/di

Api for Gaana etc is not officially Documented. But API for spotify exists , for youtube Api is there but its limited , so I use scraping for Youtube.

**Reply by MridulNegi2005** [reply]
I am glad you replied me....
But I meant to ask something else.
I know that Spotify  API exist. But it doesn't allow downloading. 
So could you create API for your project that would allow us to send requests and get a download link in return?

Same like you do in an app but this should be through https requests

**Reply by Shabinder** [reply]
Ohhk @007Gamerx  I got what are you asking , Actually I have very limited free time and  I have really no intention on working this for now. But all contributions are welcome and maybe I will work on that after I integrate Jio Saavn into my app too, its on progress.

**Reply by MridulNegi2005** [reply]
Ok thanks :)
